packages/dds/tree/src/test/simple-tree/utils.ts

```diff
@@ -43,3 +44,129 @@ export function pretty(arg: unknown): number | string {
 	}
 	return JSON.stringify(arg);
 }
+
+const schemaFactory = new SchemaFactory("test");
+
+const deepTreeNode = schemaFactory.objectRecursive("deep-tree-node", {
+	node: schemaFactory.optionalRecursive([() => DeepTreeNode]),
+	value: schemaFactory.optional(schemaFactory.number),
+});
+
+export class DeepTreeNode extends deepTreeNode {}
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth < 1) {
+		throw new Error("Depth must be greater than 0");
+	}
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode = tree;
+	let depth = 1;
+
+	while (currentNode.node !== undefined) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode.value ?? 0;
+	return { depth, value };
+}
+
+/**
+ * Update the deep tree with a new value
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
+	let currentNode = tree;
+
+	while (currentNode.node !== undefined) {
+		currentNode = currentNode.node;
+	}
+
+	currentNode.value = newValue;
+}
+
+const wideTreeNode = schemaFactory.object("wide-tree-node", {
+	values: schemaFactory.array(schemaFactory.number),
+});
+
+export class WideTreeNode extends wideTreeNode {}
```

Comment:
This is not a very safe (or recommended) pattern since exactly one of these needs to be used everywhere. Its better to avoid introducing something which using would be a bug:

```suggestion
export class WideTreeNode extends  schemaFactory.object("wide-tree-node", {
	values: schemaFactory.array(schemaFactory.number),
}) {};
```

Reply:
Updated